tests/tensor/test_basic.py

```diff
@@ -4613,19 +4613,12 @@ def core_np(x, y):
 
 @pytest.mark.parametrize(
     "ift, iff",
-    [(None, None), (7, 10), (7, None)],
+    [(None, None), (7, 10), pytest.param(7, None, marks=[pytest.mark.xfail])],
```

Comment:
Nitpick number two, this is not a standard use of xfail? The code is working correctly, usually xfail is to mark code that we know is not working correctly?

Reply:
I doubt we ever used xfail to check a function is asserting valid inputs

Reply:
Not to sound like an old man, it makes little sense to parametrize this test instead of just adding 3 assertions in a test:

```python
test_where():
  x = np.arange(10):
  
  # Test three inputs
  np.testing.assert_array_equal(pt.where(x > 5, 99, x).eval(), np.where(x, 99, x))
  
  # Test one input
  for pt_out, np_out in zip(pt.where(x > 5), np.where(x > 5):
    np.testing.assert_array_equal(pt_out.eval(), np_out)

  # Test error for invalid number of inputs
  with pytest.raises(..., match=...):
    pt.where(x > 5, 99)
```

Reply:
oh, should i not be using that to check for cases where it should be failing? (even though it is the correct behavior)

Reply:
Yeah I never saw xfail used for that and the documentation in pytest also doesn't suggest anything like it. PyTest will also treat it differently in the output, marks it as gray. Usually xfail is something we would like to eventually become just a passing condition, which this will never be